src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -1319,6 +1311,111 @@ nns_edge_create_handle (const char *id, nns_edge_connect_type_e connect_type,
     goto error;
   }
 
+  *edge_h = eh;
+  return NNS_EDGE_ERROR_NONE;
+
+error:
+  nns_edge_release_handle (eh);
+  return ret;
+}
+
+/**
+ * @brief Create edge custom handle.
+ */
+int
+nns_edge_custom_create_handle (const char *id, const char *lib_path,
+    nns_edge_node_type_e node_type, nns_edge_h * edge_h)
+{
+  int ret = NNS_EDGE_ERROR_NONE;
+  nns_edge_handle_s *eh;
+  void *custom_handle;
+  NnsEdgeCustomDef *custom_h;
+
+  if (node_type < 0 || node_type >= NNS_EDGE_NODE_TYPE_UNKNOWN) {
+    nns_edge_loge ("Invalid param, set exact node type.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!STR_IS_VALID (lib_path)) {
+    nns_edge_loge ("Invalid param, given custom lib path is invalid.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!edge_h) {
+    nns_edge_loge ("Invalid param, edge_h should not be null.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  ret = _nns_edge_create_handle (id, node_type, edge_h);
+  if (ret != NNS_EDGE_ERROR_NONE) {
+    nns_edge_loge ("Failed to create edge handle.");
+    return ret;
+  }
+  eh = (nns_edge_handle_s *) * edge_h;
+  eh->connect_type = NNS_EDGE_CONNECT_TYPE_CUSTOM;
+
+  custom_handle = dlopen (lib_path, RTLD_LAZY);
+  if (custom_handle) {
+    void *(*getCustomHandle) () =
+        (void *(*)()) dlsym (custom_handle, "nns_edge_custom_get_handle");
+    if (!getCustomHandle) {
+      nns_edge_loge ("Failed to find nns_edge_custom_get_handle: %s",
+          dlerror ());
+      return NNS_EDGE_ERROR_UNKNOWN;
+    }
+    eh->custom_h = getCustomHandle ();
+  } else {
+    nns_edge_loge ("Failed to open custom handle: %s]\n", dlerror ());
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;;
+  }
+
+  custom_h = (NnsEdgeCustomDef *) eh->custom_h;
+  ret = custom_h->nns_edge_custom_create (&eh->broker_h);
+  if (NNS_EDGE_ERROR_NONE != ret) {
+    nns_edge_loge ("Failed to create custom connection handle.");
+  }
+
+  return ret;
+}
+
+/**
+ * @brief Create edge handle.
+ */
+int
+nns_edge_create_handle (const char *id, nns_edge_connect_type_e connect_type,
+    nns_edge_node_type_e node_type, nns_edge_h * edge_h)
+{
+  int ret = NNS_EDGE_ERROR_NONE;
+  nns_edge_handle_s *eh;
+
+  if (connect_type < 0 || connect_type >= NNS_EDGE_CONNECT_TYPE_UNKNOWN) {
+    nns_edge_loge ("Invalid param, set valid connect type.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  /**
+   * @todo handle flag (receive | send)
+   * e.g., send only case: listener is unnecessary.
+   */
+  if (node_type < 0 || node_type >= NNS_EDGE_NODE_TYPE_UNKNOWN) {
```

Comment:
It seems like you defined nns_edge_create_handle() well. :100: 
It seems like using ```NNS_EDGE_CONNECT_TYPE_TCP``` is better than using ``0```. What do you think about it? :smiley_cat:

Reply:
Oh, I see. Thank you for the detailed explanation. 🙇